File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head or k == 0 : return head
        curr = head
        len = 1
        
        while curr.next :
            len+=1
            curr = curr.next
        k %= len
        curr.next = head
        cnt = len - k - 1
        temp = head
        while cnt :
            temp = temp.next
            cnt -= 1
        newHead = temp.next
        temp.next = None
        return newHead

File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        
        if(head == null || head.next == null || k == 0) return head;
        
        ListNode dummy = head;
        int len = 1;
        while(dummy.next != null){
            len++;
            dummy = dummy.next;
        }
        
        dummy.next = head;
        k = len - (k % len) - 1;
        
        while(k > 0){
            head = head.next;
            k--;
        }
        
        dummy = head.next;
        head.next = null;
        
        return dummy;
    }
}